build/guide/intro.md

```diff
@@ -0,0 +1,173 @@
+---
+title: 1. Introduction
+description: An introduction to the Docker Build guide
+keywords: build, buildkit, buildx, guide, tutorial, introduction
+---
+
+{% include_relative nav.html selected="1" %}
+
+The starting resources for this guide includes a simple Go project and a
+Dockerfile. From this starting point, the guide illustrates various ways that
+you can improve how you build the application with Docker.
+
+## Environment setup
+
+To follow this guide:
+
+1. Install [Docker Desktop or Docker Engine](../../get-docker.md)
+2. Clone or create a new repository from the
+   [application example on GitHub](https://github.com/dvdksn/buildme)
+
+## The application
+
+The example project for this guide is a client-server application for
+translating messages to a fictional language.
+
+Here’s an overview of the files included in the project:
+
+```text
+.
+├── Dockerfile
+├── cmd
+│   ├── client
+│   │   ├── main.go
+│   │   ├── request.go
+│   │   └── ui.go
+│   └── server
+│       ├── main.go
+│       └── translate.go
+├── go.mod
+└── go.sum
+```
+
+The `cmd/` directory contains the code for the two application components:
+client and server. The client is a user interface for writing, sending, and
+receiving messages. The server receives messages from clients, translates them,
+and sends them back to the client.
+
+## The Dockerfile
+
+A Dockerfile is a text document in which you define the build steps for your
+application. You write the Dockerfile in a domain-specific language, called the
+Dockerfile syntax.
+
+Here's the Dockerfile used as the starting point for this guide:
+
+```dockerfile
+# syntax=docker/dockerfile:1
+FROM golang:1.20.3-alpine
+WORKDIR /src
+COPY . .
+RUN go mod download
+RUN go build -o /bin/client ./cmd/client
+RUN go build -o /bin/server ./cmd/server
+ENTRYPOINT [ "/bin/server" ]
+```
+
+Here’s what this Dockerfile does:
+
+1. `# syntax=docker/dockerfile:1`
+
+   This comment is a
+   [Dockerfile parser directive](../../engine/reference/builder.md#parser-directives).
+   It specifies which version of the Dockerfile syntax to use. This file uses
+   the `dockerfile:1` syntax which is best practice: it ensures that you have
+   access to the latest Docker build features.
+
+2. `FROM golang:1.20.3-alpine`
+
+   The `FROM` instruction uses version 1.20.3 of the `golang` official image, on
+   Alpine Linux, as a base image.
```

Comment:
Ah good point, actually, we could add a site-level variable and reference that. I think it would work (except images would obviously not be updated, but that's probably fine), i.e.

```yaml
# config.yml
example_go_version: "1.20.3"
```

And then in our docs examples, reference the variable:

```dockerfile
FROM golang:{{site.example_go_version}}-alpine
```

Reply:
Agree, I think we could either set `latest` or `1.20` and explain that `1.20` means it takes the very latest patch if user wants to be bug free when iterating with his project without worrying to update his Dockerfile. For `latest` warns user this is not recommended as it would take the latest stable and might break his workflow.